Dropdown/src/Dropdown.jsx

```diff
@@ -231,69 +243,134 @@ class Dropdown extends Component {
+                       className={this.getOptionClassName(option, index)}>{option.label}</li>;
         });
+        return options;
+    }
+
+    getOptionClassName(option, index) {
+        const {props, state} = this;
+        const currentIndex = this.getCurrentIndex();
+        const isCurrentIndex = index === currentIndex;
+        const isPreselected = props.value != null && (option.value === props.value && state.closestValue === null && currentIndex < 0);
+        const isSearchResult = state.closestValue != null && (option.value === state.closestValue);
+        const selected = isCurrentIndex || isSearchResult || isPreselected;
+        return selected ? "dnn-dropdown-option selected" : "dnn-dropdown-option";
+    }
+
+    isSelectedItem(index) {
+        return index == this.state.currentIndex;
+    }
+
+
+    addOptionRef(option) {
+        if (option) {
+            this.selectedOptionElement = option;
+        }
+    }
+
+
+    getOption(index) {
+        const options = this.optionItems;
+        return options && options[index] !== undefined ? options[index] : null;
+    }
+
+    /* eslint-disable react/no-danger */
+    render() {
+        const {props, state} = this;
+        const options = this.initOptions();
         return (
             <div className={this.getClassName()} style={props.style}>
-                <div className={"collapsible-label" + this.getIsMultiLineLabel()} 
-                    onClick={this.toggleDropdown.bind(this)} title={this.props.title}>
+                <div className={"collapsible-label" + this.getIsMultiLineLabel()}
+                     onClick={this.toggleDropdown.bind(this)} title={this.props.title}>
                     {this.getDropdownLabel()}
                 </div>
                 <input
                     type="text"
                     onChange={this.onDropdownSearch.bind(this)}
-                    ref="dropdownSearch"
+                    ref={(input) => this.dropdownSearch = input}
                     value={this.state.dropdownText}
                     onKeyDown={this.onKeyDown.bind(this)}
-                    style={{ position: "absolute", opacity: 0, pointerEvents: "none", width: 0, height: 0, padding: 0, margin: 0 }}
+                    style={{
+                        position: "absolute",
+                        opacity: 0,
+                        pointerEvents: "none",
+                        width: 0,
+                        height: 0,
+                        padding: 0,
+                        margin: 0
+                    }}
                     aria-label="Search"
-                    />
-                {props.withIcon && <div className="dropdown-icon" dangerouslySetInnerHTML={{ __html: ArrowDownIcon }} onClick={this.toggleDropdown.bind(this)}></div>}
+                />
+                {props.withIcon && <div className="dropdown-icon" dangerouslySetInnerHTML={{__html: ArrowDownIcon}}
```

Comment:
Is there a way to reduce the usage of `dangerouslySetInnerHTML`?

Reply:
seems to be the only way to inject icons AFAIK.

Reply:
This is the way the icons are injected when they are loaded with the _raw-loader_ . There is another probably cleaner way to use icons by loading them using the _svg-sprite-loader_ that has bee used in other projects. Unfortunately, most of the icons in the common components are loaded this way, and changing it would be a breaking change and is something that needs to be evaluated